packages/core/src/utils/StorageUtil.ts

```diff
@@ -280,5 +297,243 @@ export const StorageUtil = {
     } catch {
       console.info('Unable to remove connected namespace')
     }
+  },
+  getBalanceCache() {
+    let cache: Record<string, { timestamp: number; balance: BlockchainApiBalanceResponse }> = {}
+    try {
+      const result = SafeLocalStorage.getItem(SafeLocalStorageKeys.PORTFOLIO_CACHE)
+      cache = result ? JSON.parse(result) : {}
+    } catch {
+      console.info('Unable to get balance cache')
+    }
+
+    return cache
+  },
```

Comment:
do we need this if we always fetch by caip address?

Reply:
its used to get the item from localStorage and then its filtered by caipaddress